File: backend/routes/resultados.py
from flask import Blueprint, jsonify
from db import get_connection
from auth_helpers import login_requerido
import json

resultados_bp = Blueprint('resultados', __name__)


@resultados_bp.route('/resultados', methods=['GET'])
@login_requerido
def listar_resultados(usuario_id):
    """Retorna todos los análisis del usuario ordenados del más reciente al más antiguo."""
    conn = get_connection()
    try:
        with conn.cursor() as cur:
            cur.execute(
                '''SELECT id, tipo_estudio, fecha_estudio, valores, resumen_ia, estado, creado_en
                   FROM resultados
                   WHERE usuario_id = %s
                   ORDER BY fecha_estudio DESC, creado_en DESC''',
                (usuario_id,)
            )
            rows = cur.fetchall()

        resultados = []
        for r in rows:
            resultados.append({
                'id':            r['id'],
                'tipo_estudio':  r['tipo_estudio'],
                'fecha_estudio': str(r['fecha_estudio']),
                'valores':       json.loads(r['valores']) if isinstance(r['valores'], str) else r['valores'],
                'resumen_ia':    r['resumen_ia'],
                'estado':        r['estado'],
                'creado_en':     str(r['creado_en']),
            })

        return jsonify({'resultados': resultados}), 200

    except Exception as e:
        return jsonify({'error': 'Error al obtener los resultados.'}), 500
    finally:
        conn.close()


@resultados_bp.route('/resultados/<int:resultado_id>', methods=['GET'])
@login_requerido
def obtener_resultado(usuario_id, resultado_id):
    """Retorna un análisis específico verificando que pertenezca al usuario."""
    conn = get_connection()
    try:
        with conn.cursor() as cur:
            cur.execute(
                '''SELECT id, tipo_estudio, fecha_estudio, valores, resumen_ia, estado
                   FROM resultados
                   WHERE id = %s AND usuario_id = %s''',
                (resultado_id, usuario_id)
            )
            r = cur.fetchone()

        if not r:
            return jsonify({'error': 'Análisis no encontrado.'}), 404

        resultado = {
            'id':            r['id'],
            'tipo_estudio':  r['tipo_estudio'],
            'fecha_estudio': str(r['fecha_estudio']),
            'valores':       json.loads(r['valores']) if isinstance(r['valores'], str) else r['valores'],
            'resumen_ia':    r['resumen_ia'],
            'estado':        r['estado'],
        }
        return jsonify({'resultado': resultado}), 200

    except Exception as e:
        return jsonify({'error': 'Error al obtener el resultado.'}), 500
    finally:
        conn.close()


@resultados_bp.route('/resultados/<int:resultado_id>', methods=['DELETE'])
@login_requerido
def eliminar_resultado(usuario_id, resultado_id):
    """Elimina un análisis verificando que pertenezca al usuario."""
    conn = get_connection()
    try:
        with conn.cursor() as cur:
            cur.execute(
                'DELETE FROM resultados WHERE id = %s AND usuario_id = %s',
                (resultado_id, usuario_id)
            )
            eliminados = cur.rowcount
        conn.commit()

        if eliminados == 0:
            return jsonify({'error': 'Análisis no encontrado o sin permiso.'}), 404

        return jsonify({'ok': True}), 200

    except Exception as e:
        conn.rollback()
        return jsonify({'error': 'Error al eliminar el análisis.'}), 500
    finally:
        conn.close()

File: backend/api/resultados.php
<?php
require_once 'config.php';

$method = $_SERVER['REQUEST_METHOD'];
$usuario_id = verifyToken();
$resultado_id = $_GET['id'] ?? null;

if ($method === 'GET' && !$resultado_id) {
    $db = getDB();
    $stmt = $db->prepare('SELECT id, tipo_estudio, fecha_estudio, valores, resumen_ia, estado, creado_en FROM resultados WHERE usuario_id = ? ORDER BY fecha_estudio DESC, creado_en DESC');
    $stmt->bind_param('i', $usuario_id);
    $stmt->execute();
    $rows = $stmt->get_result()->fetch_all(MYSQLI_ASSOC);
    foreach ($rows as &$r) {
        $r['valores'] = json_decode($r['valores'], true);
        $r['fecha_estudio'] = (string)$r['fecha_estudio'];
        $r['creado_en'] = (string)$r['creado_en'];
    }
    jsonResponse(['resultados' => $rows]);
}

if ($method === 'GET' && $resultado_id) {
    $db = getDB();
    $stmt = $db->prepare('SELECT id, tipo_estudio, fecha_estudio, valores, resumen_ia, estado FROM resultados WHERE id = ? AND usuario_id = ?');
    $stmt->bind_param('ii', $resultado_id, $usuario_id);
    $stmt->execute();
    $r = $stmt->get_result()->fetch_assoc();
    if (!$r) jsonResponse(['error' => 'Análisis no encontrado.'], 404);
    $r['valores'] = json_decode($r['valores'], true);
    $r['fecha_estudio'] = (string)$r['fecha_estudio'];
    jsonResponse(['resultado' => $r]);
}

if ($method === 'DELETE' && $resultado_id) {
    $db = getDB();
    $stmt = $db->prepare('DELETE FROM resultados WHERE id = ? AND usuario_id = ?');
    $stmt->bind_param('ii', $resultado_id, $usuario_id);
    $stmt->execute();
    if ($db->affected_rows === 0) jsonResponse(['error' => 'Análisis no encontrado.'], 404);
    jsonResponse(['ok' => true]);
}

jsonResponse(['error' => 'Ruta no encontrada'], 404);
